tutorials/run-laravel-on-appengine-standard/index.md

```diff
@@ -89,19 +102,149 @@ from laravel.com.
 
-1.  Finally, remove the `beyondcode/laravel-dump-server` composer dependency. This is a
-    fix for an error which happens as a result of Laravel's caching in
-    `bootstrap/cache/services.php`.
+    In this tutorial we will actually be hard-coding storage path in later 
+    step, so the `APP_STORAGE` configuration remains only for testing locally. 
+    This is because `env()` function should not be called after caching Laravel 
+    configuration, which we will be doing later.
+    
+    Still, don't skip this step, and don't modify the characters within 
+    `useStoragePath(...)`, as they will be matched and substituted by `sed` 
+    later on.
+
+1.  Modify `public/index.php` by adding the following block of code after the 
+    definition of `'LARAVEL_START'`. This helps to avoid the error when Laravel 
+    tries to write into storage directory that does not exist.
 
-        composer remove --dev beyondcode/laravel-dump-server
-        
-    If you're using Laravel 6, remove `facade/ignition` instead:
+    We have to do this at runtime because each instance in App Engine gets its 
+    own writable in-memory `/tmp` directory. Creating these directories during 
+    deployment would be a waste of time, because the runtime version of `/tmp` 
+    is not yet mounted at that time.
+
+        # [START] Add the following block to `public/index.php`
+        /*
+        |--------------------------------------------------------------------------
+        | GCP App Engine Writable Directories
+        |--------------------------------------------------------------------------
+        |
+        | Before we start up Laravel, let's check if we are running in GCP App Engine,
+        | and, if so, make sure that the Laravel's storage directory structure is
+        | present.
+        |
+        */
+
+        if (getenv('GCP_APP_ENGINE_LARAVEL')
+            && !file_exists('/tmp/.dirs_created')) {
+            foreach (['/tmp/app/public',
+                        '/tmp/framework/cache/data',
+                        '/tmp/framework/sessions',
+                        '/tmp/framework/testing',
+                        '/tmp/framework/views',
+                        '/tmp/logs'] as $tmpdir) {
+                if (!file_exists($tmpdir)) {
+                    mkdir($tmpdir, 0755, true);
+                }
+            }
+            touch('/tmp/.dirs_created');
+        }
+        # [END]
     
-        composer remove --dev facade/ignition
+    This code is where the `GCP_APP_ENGINE_LARAVEL` environment variable from 
+    `app.yaml` comes into play. This avoids running this code in local 
+    development environment.
+
+    If your code writes into any storage directory, you should make sure to 
+    create the necessary directory structure here.
+
+1.  Modify `composer.json` file by adding a few scripts scripts. Here is an 
+    example result, with the additions detailed below:
+
+        "scripts": {
+            "post-autoload-dump": [
```

Comment:
Rather than have the users add the `gcp-build` script block manually, consider displaying them in `composer` command format:

```sh
composer config scripts.gcp-build "sed -i -e \"s|env('APP_STORAGE', base_path() . '/storage')|'/tmp'|g\" bootstrap/app.php && mkdir -p /tmp/framework/views && mv .env.gae .env && php artisan config:cache && rm -f .env && php artisan route:cache"
```

Reply:
Everyday you learn something new, such as Composer's `config` command :) I'll push edits after I run them locally tomorrow.